### PR TITLE
✨ Implement rough support for wallet connect v2

### PR DESCRIPTION
**Pull request by williamchong**
TODO:
- Support parsing of chain id and methods in session proposal 
- Improve signing information and UX

### CODE REVIEW

File: app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    listenToRequests() {
+      self.connector.on('session_proposal', self.handleNewSessionRequest)
+      self.connector.on('session_request', self.handleNewCallRequest)
+      self.connector.on('session_delete', self.handleDisconnect)
+    },
+  }))
+  .actions(self => ({
+    createClient: flow(function * (){
+      const client = yield SignClient.init({
+        projectId: self.getConfig('WALLET_CONNECT_PROJECT_ID', 'd36730c770de6ccc7db8591473e97f6f'),
```
Comment (nwingt) [inline app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts:287]: Should remove default value?
```suggestion
        projectId: self.getConfig('WALLET_CONNECT_PROJECT_ID'),
```

File: app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+        metadata: {
+          name: 'LikerLand',
+          description: 'Liker Land App',
+          url: 'https://like.co',
+          icons: ['https://like.co/logo.png'],
```
Comment (nwingt) [inline app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts:292]:
```suggestion
          name: 'Liker Land',
          description: 'Liker Land App',
          url: 'https://like.co',
          icons: ['https://liker.land/logo.png'],
```

File: app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+      })
+    },
+    rejectCallRequest(response: {
+      id: number,
+      topic: string
+      error: any
+    }) {
+      return self.connector.respond({
+        topic: response.topic,
+        response:formatJsonRpcError(response.id, getSdkError('USER_REJECTED_METHODS').message)
```
Comment (nwingt) [inline app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts:77]:
```suggestion
        response: formatJsonRpcError(response.id, getSdkError('USER_REJECTED_METHODS').message)
```

File: app/models/deep-link-handle-store/deep-link-handle-store.ts [hunk 4]
```diff
@@ -145,8 +145,20 @@ const BaseModel = types
...
+            else if (url.includes('wc?')) {
+              [, walletConnectURI = ''] = url.split('wc?')
+            }
+            else if (url.startsWith('wc:')) walletConnectURI = url
+            if (walletConnectURI.startsWith('uri=')) {
+              // com.oice://wc/wc?uri=wc%3A327ac60...
+              walletConnectURI = walletConnectURI.replace('uri=', '')
+              walletConnectURI = walletConnectURI.split('&')[0]
+              walletConnectURI = decodeURIComponent(walletConnectURI)
+              console.log(walletConnectURI);
```
Comment (nwingt) [inline app/models/deep-link-handle-store/deep-link-handle-store.ts:160]: Please remove log

File: app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+import { withCurrentUser, withEnvironment, withNavigationStore } from '../extensions'
+
+function isSessionRequest(method: string) {
+  return method.includes('proposal')
+}
+
+export const WalletConnectV2ClientModel = types
+  .model('WalletConnectV2Client')
+  .volatile(() => ({
+    isMobile: false,
```
Comment (nwingt) [inline app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts:21]: Can be removed if not used?

File: app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts (hunk 1 above)
Comment (williamchong) [reply app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts:286]: Need some help on why `WALLET_CONNECT_PROJECT_ID` is not get from env here, other comments are updated

**Review by nwingt — APPROVED** [review]

File: app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts (hunk 1 above)
Comment (nwingt) [reply app/models/wallet-connect-v2-client/wallet-connect-v2-client.ts:286]: I can see it
image

**Pull request merged**